src/EFCore.PG/Query/NpgsqlSqlExpressionFactory.cs

```diff
@@ -391,14 +392,42 @@ private SqlExpression ApplyTypeMappingOnSqlBinary(SqlBinaryExpression binary, Re
             }
         }
 
-        return base.ApplyTypeMapping(binary, typeMapping);
+        if (binary.OperatorType is ExpressionType.GreaterThan or ExpressionType.LessThan
+                or ExpressionType.GreaterThanOrEqual or ExpressionType.LessThanOrEqual
+            && binary.Left is PostgresRowValueExpression leftRowValue
+            && binary.Right is PostgresRowValueExpression rightRowValue)
+        {
+            Check.DebugAssert(leftRowValue.Values.Count == rightRowValue.Values.Count, "Row value count mismatch in comparison");
+
+            var count = leftRowValue.Values.Count;
+            var updatedLeftValues = new SqlExpression[count];
+            var updatedRightValues = new SqlExpression[count];
+
+            for (var i = 0; i < count; i++)
+            {
+                var updatedElementBinaryExpression =
```

Comment:
👍🏻

Reply:
The emoji doesn't render here well :rofl: Hopefully it's positive...

Reply:
I see you what you did here. 😉

Reply:
Ah OK :)